File: src/features/typing/data/go/symmetric_tree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isSymmetric(root *TreeNode) bool {
	var mirror func(a, b *TreeNode) bool
	mirror = func(a, b *TreeNode) bool {
		if a == nil || b == nil {
			return a == b
		}
		return a.Val == b.Val && mirror(a.Left, b.Right) && mirror(a.Right, b.Left)
	}
	return mirror(root, root)
}

File: src/features/typing/data/dart/symmetric_tree.dart
class Solution {
  bool isSymmetric(TreeNode? root) {
    bool mirror(TreeNode? a, TreeNode? b) {
      if (a == null || b == null) {
        return a == b;
      }
      return a.val == b.val && mirror(a.left, b.right) && mirror(a.right, b.left);
    }
    return mirror(root, root);
  }
}
